core/src/main/java/io/parsingdata/metal/token/Until.java

```diff
@@ -106,9 +107,10 @@ public Until(final String name, final ValueExpression initialSize, final ValueEx
     }
 
     private Trampoline<Optional<Environment>> parseSlice(String scope, Environment environment, BigInteger currentSize, BigInteger stepSize, BigInteger maxSize, Encoding encoding, Slice slice) {
-        return terminator
-            .parse(scope, currentSize.compareTo(ZERO) == 0 ? environment : environment.add(new ParseValue(name, this, slice, encoding)).seek(environment.offset.add(currentSize)), encoding)
-            .map(nextEnvironment -> complete(() -> success(nextEnvironment)))
+        return (currentSize.compareTo(ZERO) == 0 ? Optional.of(environment) : environment.add(new ParseValue(name, this, slice, encoding)).seek(environment.offset.add(currentSize)))
```

Comment:
I'd prefer if you just turn the zero-check into a guard clause here, instead of "cramming" it into the single statement. (I know you're a fan... 😁) The creation of the optional environment seems a bit _forced_.

Reply:
Returns only, right?

Reply:
Agreed that the `Optional.of(environment)` is dodgy. But it either explodes into a really verbose `if` (with `isPresent()`, `get()`, etc.) and even removes the first `map()` or it's like this...